src/kernel/crane_constructor/hook_chooser/hook.rs

```diff
@@ -0,0 +1,179 @@
+use crate::kernel::storage::storage::Storage;
+use crate::kernel::storage::storage::Value;
+use crate::Param_to_compare;
+use std::io;
+pub struct Hook{
+    // Vec<Vec<String> Крюки<Характеристики>
+
+    pub hook: Vec<String>,
+    pub bearing: String
+}
+
+
+
+impl Hook{
+
+    pub fn new(_param_comp: Param_to_compare, hooks_storage: &mut Storage) -> Self {
+
+        println!("{}",_param_comp._hook_type);
+        //Выбор крюка
+        let res_hook = Self::hook_select(Self::weight_check(&_param_comp, hooks_storage));
+
```

Comment:
If you have an empty lines in your code blocks,
then you try to highlighting a different functionality.
So such code blocks must be splitted into different classes,
or at least into different methods